lib/signet/oauth_2/client.rb

```diff
@@ -1020,10 +1046,13 @@ def fetch_access_token options = {}
           content_type = response.header[:content_type]
         end
 
-        return ::Signet::OAuth2.parse_credentials body, content_type if status == 200
-
         message = "  Server message:\n#{response.body.to_s.strip}" unless body.to_s.strip.empty?
-        if [400, 401, 403].include? status
+
+        if status == 200
+          parsed_response = ::Signet::OAuth2.parse_credentials body, content_type
+          parsed_response["granted_scopes"] = parsed_response.delete("scope") if parsed_response
+          parsed_response
```

Comment:
should this be a return statement?

Reply:
The `return` keyword is not necessary, and is generally idiomatically omitted, since the value of the last expression in a method (which in this case is whatever clause in the if-elsif-else is selected),  is implicitly the return value.

That said, this construct would be better turned around, because currently the return value is buried in the middle of the method and thus isn't obvious. (It also invites a subtle bug where someone might inadvertently add another statement after this if-elsif-else clause, thus causing something different to be returned from the method.) It would be better for the else clause to be `elsif status != 200`, and then put this parsed_response stuff at the very end of the method, so that the return value is positioned in the obvious place at the end.